### PR TITLE
Add detect/tests for prefetch

### LINKED ISSUE #2536: Rel Prefetch Support?
Any chance we could get a detect for rel prefetch? It'd be super-helpful for WebKit browsers. There's a [bug out for it](https://bugs.webkit.org/show_bug.cgi?id=194539) now, but no movement yet.

### PR DESCRIPTION
**Pull request by HeroProtagonist**
Closes #2536

- Made a detect for prefetch based on https://github.com/w3c/preload/issues/7
- Added general test and caniuse test

### CODE REVIEW

**Comment by robin-drexler** [comment]
Hey, just stumbled upon this. This check may exclude IE11 which appears to [support prefetch](https://caniuse.com/link-rel-prefetch), but [not relList](https://caniuse.com/rellist) :( 

This isn’t an issue for preload because IE supports neither.

**Comment by HeroProtagonist** [comment]
Do any of the detects do browser detection for a case like this?
I see a User Agent parser here, but not in source code

https://github.com/Modernizr/Modernizr/blob/94592f279a410436530c7c06acc42a6e90c20150/test/browser/integration/caniuse.js#L4

File: feature-detects/rel/prefetch.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*!
```
Comment (rejas) [inline feature-detects/rel/prefetch.js:1]: creating a REl directory for this test only seems coutnerintuitive. isnt there a better palce for this?

**Comment by rejas** [comment]
yeah, that UAparser is onyl used during testing of Modernizr to check against the caniuse database. It is generally not advised to use it in normal feature tests.

**Comment by HeroProtagonist** [comment]
> yeah, that UAparser is onyl used during testing of Modernizr to check against the caniuse database. It is generally not advised to use it in normal feature tests.

Looks like `navigator.userAgent` is used in some detections. Maybe that is a better option

https://github.com/Modernizr/Modernizr/search?q=navigator.userAgent

File: feature-detects/rel/prefetch.js (hunk 1 above)
Comment (HeroProtagonist) [reply feature-detects/rel/prefetch.js:1]: I can move it, just was thinking a follow up pr would be for `rel=preload`

**Comment by rejas** [comment]
Yes, that sounds reasonable :-)

File: feature-detects/rel/prefetch.js (hunk 1 above)
Comment (rejas) [reply feature-detects/rel/prefetch.js:1]: Then I propose putting it in a "link" subdirectory, since it is a test on the link element ...

**Comment by HeroProtagonist** [comment]
@rejas Added ie11 check using `document.documentMode` which is used in other parts of the code for similar IE specific detection

File: feature-detects/rel/prefetch.js (hunk 1 above)
Comment (HeroProtagonist) [reply feature-detects/rel/prefetch.js]: done!

**Comment by rejas** [comment]
Thanks for the PR and your persistence :-) Merged!

**Pull request merged**